apps/crn-server/src/data-providers/contentful/interest-groups.data-provider.ts

```diff
@@ -0,0 +1,174 @@
+import {
+  FetchGroupOptions,
+  GroupDataObject,
+  GroupTools,
+  GroupRole,
+  GroupLeader,
+  ListGroupDataObject,
+} from '@asap-hub/model';
+import {
+  GraphQLClient,
+  FETCH_INTEREST_GROUPS,
+  FETCH_INTEREST_GROUP_BY_ID,
+  FetchInterestGroupByIdQuery,
+  FetchInterestGroupByIdQueryVariables,
+  FetchInterestGroupsQuery,
+  FetchInterestGroupsQueryVariables,
+  InterestGroupsOrder,
+  InterestGroupsFilter,
+  Teams,
+} from '@asap-hub/contentful';
+
+import { InterestGroupDataProvider } from '../types';
+import {
+  parseContentfulGraphqlCalendarToResponse,
+  parseInterestGroupLeader,
+} from '../../entities';
+import { parseContentfulGraphQlUsers } from './users.data-provider';
+import { parseContentfulGraphQlTeams } from './teams.data-provider';
+
+export type InterestGroupItem = NonNullable<
+  NonNullable<FetchInterestGroupByIdQuery['interestGroups']>
+>;
+
+export class InterestGroupContentfulDataProvider
+  implements InterestGroupDataProvider
+{
+  constructor(private contentfulClient: GraphQLClient) {}
+
+  async fetchById(id: string): Promise<GroupDataObject | null> {
+    const { interestGroups } = await this.contentfulClient.request<
+      FetchInterestGroupByIdQuery,
+      FetchInterestGroupByIdQueryVariables
+    >(FETCH_INTEREST_GROUP_BY_ID, { id });
+
+    if (!interestGroups) {
+      return null;
+    }
+
+    return parseGraphQLInterestGroup(interestGroups);
+  }
+
+  async fetch(options: FetchGroupOptions): Promise<ListGroupDataObject> {
+    const { filter, search, take = 20, skip = 0 } = options;
+
+    const where: InterestGroupsFilter = {};
+    const words = (search || '').split(' ').filter(Boolean); // removes whitespaces
+
+    if (words.length) {
+      const filters: InterestGroupsFilter[] = words.reduce(
+        (acc: InterestGroupsFilter[], word: string) =>
+          acc.concat([
+            {
+              OR: [
+                { name_contains: word },
+                { description_contains: word },
+                { tags_contains_all: [word] },
+              ],
+            },
+          ]),
+        [],
+      );
+      where.AND = filters;
+    }
+
+    if (filter && filter.active !== undefined) {
```

Comment:
The filter by userId and by teamId is missing and these filters are being used here:

- By Team Id: https://github.com/yldio/asap-hub/blob/master/apps/crn-server/src/routes/teams.route.ts#L77 / https://github.com/yldio/asap-hub/blob/master/apps/crn-frontend/src/network/teams/groups/api.ts#L10
- By User Id: https://github.com/yldio/asap-hub/blob/master/apps/crn-server/src/routes/user.route.ts#L96 / https://github.com/yldio/asap-hub/blob/master/apps/crn-frontend/src/network/users/groups/api.ts#L9